x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/execution/search/extractor/FieldHitExtractor.java

```diff
@@ -142,6 +173,13 @@ private Object unwrapMultiValue(Object values) {
                 }
             }
         }
+        if (dataType == DataType.GEO_SHAPE) {
+            try {
+                return new GeoShape(values);
+            } catch (IOException ex) {
+                throw new SqlIllegalArgumentException("Cannot read geo_shape value (returned by [{}])", fieldName);
```

Comment:
Please, include the actual value that cannot be read in the error message, to be consistent with the rest of the error messages.